File: tests/test_cases.java

import java.awt.Robot;

/*
 * TODO: set correct path of zas.class before compiling: 
 *
 *  export CLASSPATH=$CLASSPATH:/mnt/sda5/zyw/work/zas/wrapper/java
 */

public class test_cases {

	public static void main(String[] args) throws Exception {

    zas mz = (zas)Class.forName("zas").newInstance();

    mz.login("localhost",3306,"root","123","");

    mz.prepare("select id,name,price,size from test_db.test_tbl where id<:f1<int>");

    while (true) {

      mz.insertInt(10);

      while (!mz.isEof()) {
        Integer id = 0;
        String name = "" ;
        Double price = 0.0;
        Long size = 0L;

        id = mz.fetchInt();
        name = mz.fetchStr();
        price = mz.fetchDouble();
        size = mz.fetchLong();
        System.out.println(id + ": name: " + name + ", point: " + 
          price + ", size: " + size + "\n");
      }

      //System.out.println("ok!\n");

    }

  }
}

File: src/zas/tests/test_cases.py
#!/usr/bin/python3 -b

import sys

# TODO: set correct path of the zas.py 
sys.path.append("/mnt/sda5/zyw/work/zas/wrapper")

from python.zas import *

def main():

  mz = zas()

  if mz.login("localhost",3306,"root","123","")!=0 :
    print("login fail\n")
    exit(-1)

  if mz.prepare('select *from test_db.test_tbl where id <:f1<int>')!=0 :
    print("prepare fail\n")
    exit(-1)

  if mz.insert_int(10)!=0 :
    exit(-1)

  while mz.is_eof()==0 :
    id = mz.fetch_int()
    name = mz.fetch_str()
    price = mz.fetch_double()
    size = mz.fetch_long()
    print("{0}: name: {1}, point: {2}, size: {3}\n".format(id,name,price,size))
  
  print("ok!\n")


if __name__ == "__main__":
  main()

